### PR TITLE
=str more robust dispatcher checking in FileSource specs

### PR DESCRIPTION
**Pull request by ktoso**
More stable way to test for proper dispatcher being used in tests introduced in https://github.com/akka/akka/pull/17211

### CODE REVIEW

File: akka-stream-tests/src/test/scala/akka/stream/io/SynchronousFileSinkSpec.scala [hunk 1]
```diff
@@ -93,8 +100,9 @@ class SynchronousFileSinkSpec extends AkkaSpec(StreamTestKit.UnboundedMailboxCon
         try {
           Source(() ⇒ Iterator.continually(TestByteStrings.head)).runWith(SynchronousFileSink(f))(mat)
 
-          val ref = Await.result(sys.actorSelection("/user/$a/flow-1-2*").resolveOne(), timeout.duration)
-          ref.asInstanceOf[RepointableActorRef].underlying.asInstanceOf[ActorCell].dispatcher.id should ===("akka.stream.default-file-io-dispatcher")
+          mat.asInstanceOf[ActorFlowMaterializerImpl].supervisor.tell(StreamSupervisor.GetChildren, testActor)
+          val ref = expectMsgType[Children].children.find(_.path.toString contains "File").get
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/io/SynchronousFileSinkSpec.scala]: ah, that was also a problem with the wildcard

Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/io/SynchronousFileSinkSpec.scala]: Yeah, the order of actors is not deterministic in there of course :)

File: akka-stream-tests/src/test/scala/akka/stream/io/SynchronousFileSourceSpec.scala [hunk 2]
```diff
@@ -165,8 +174,9 @@ class SynchronousFileSourceSpec extends AkkaSpec(StreamTestKit.UnboundedMailboxC
       try {
         SynchronousFileSource(manyLines).runWith(Sink.ignore)(mat)
 
-        val ref = Await.result(sys.actorSelection("/user/$a/flow-*").resolveOne(), timeout.duration)
-        ref.asInstanceOf[RepointableActorRef].underlying.asInstanceOf[ActorCell].dispatcher.id should ===("akka.stream.default-file-io-dispatcher")
+        mat.asInstanceOf[ActorFlowMaterializerImpl].supervisor.tell(StreamSupervisor.GetChildren, testActor)
```
Comment (patriknw) [inline akka-stream-tests/src/test/scala/akka/stream/io/SynchronousFileSourceSpec.scala]: what if the stream is done, and shutdown before this point?

**Comment by ktoso** [comment]
(stopped build manually, needs rebase)

File: akka-stream-tests/src/test/scala/akka/stream/io/SynchronousFileSourceSpec.scala (hunk 2 above)
Comment (rkuhn) [inline akka-stream-tests/src/test/scala/akka/stream/io/SynchronousFileSourceSpec.scala]: yes, this needs a probe instead of Sink.ignore

Comment (ktoso) [inline akka-stream-tests/src/test/scala/akka/stream/io/SynchronousFileSourceSpec.scala]: You're right, this is still racy - fixing

**Comment by rkuhn** [comment]
LGTM otherwise

**Comment by ktoso** [comment]
Rebased the things, merging.

**Comment by drewhk** [comment]
LGTM

**Pull request merged**